src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketPRBranchNameDecorator.java

```diff
@@ -0,0 +1,37 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.Branch;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.Revision;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import org.jenkinsci.plugins.gitclient.GitClient;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class BitbucketPRBranchNameDecorator extends GitSCMExtension {
+
+    private final BitbucketPullRequestSCMHead pullRequestSCMHead;
+
+    public BitbucketPRBranchNameDecorator(BitbucketPullRequestSCMHead pullRequestSCMHead) {
+        this.pullRequestSCMHead = pullRequestSCMHead;
+    }
+
+    @Override
+    public void populateEnvironmentVariables(GitSCM scm, Map<String, String> env) {
+        env.put(GitSCM.GIT_BRANCH, pullRequestSCMHead.getOriginName());
```

Comment:
I can see how this solves the issue, but do you know why `GIT_BRANCH` is being set to the wrong value in the first place?
I'm wondering because I don't think this just affects pull request branches- [JENKINS-73502](https://issues.jenkins.io/browse/JENKINS-73502) is also based on an incorrectly set GIT_BRANCH and I think that also affects branch builds.
I'm happy enough with this addressing this bug in this way but I sort of expect this problem isn't solely for pull request heads, and if we may need a more general solution.

Reply:
> I can see how this solves the issue, but do you know why GIT_BRANCH is being set to the wrong value in the first place?

Yes, it uses the `name` of the `head` which is not what we want for PRs, (see comment below for elaboration)

> 'm wondering because I don't think this just affects pull request branches- [JENKINS-73502](https://issues.jenkins.io/browse/JENKINS-73502) is also based on an incorrectly set GIT_BRANCH and I think that also affects branch builds.

Correct, but different cause. This cause is because we changed what `getName` returns. They are maybe related, but not the same cause.